src/devices/src/virtio/mmio.rs

```diff
@@ -304,20 +195,28 @@ impl MmioDevice {
             }
             _ if status == 0 => {
                 if self.device_activated {
-                    match self.device.reset() {
-                        Some((_interrupt_evt, mut queue_evts)) => {
-                            self.device_activated = false;
-                            self.queue_evts.append(&mut queue_evts);
+                    let mut device_activated = true;
+                    let mut device_status = self.device_status;
+
+                    match self.locked_device().reset() {
+                        Some((_interrupt_evt, mut _queue_evts)) => {
```

Comment:
nit: let's not forget to remove `queue_evts` from the return value of `reset()`. With this new architecture they don't need to be returned, as the device owns them.